ecdf_vat_report/models/ecdf_vat_report.py

```diff
@@ -0,0 +1,588 @@
+                                 subkpis_filter,
+                                 get_additional_move_line_filter,
+                                 get_additional_query_filter,
+                                 locals_dict,
+                                 aml_model,
+                                 no_auto_expand_accounts)
+
+
+class EcdfManualInput(models.Model):
+    _name = 'ecdf.manual.input'
+
+    report_id = fields.Many2one('vat.report', string="Report")
+    name = fields.Char('Name', required=True)
+    description = fields.Char('Description')
+    value = fields.Float('Value')
+
+
+class EcdfManualData(models.Model):
+    _name = 'ecdf.manual.data'
+
+    name = fields.Char('Name', required=True)
+    description = fields.Char('Description')
+
+
+class VatReport(models.Model):
+    '''
+    Editable VAT report instance, able to create a eCDF XML file
+    '''
+    _inherit = "account.common.report"
+    _description = 'eCDF VAT Report'
+    _name = 'vat.report'
+    _transient = False
+
+    name = fields.Char(string='Name', required=True)
+    description = fields.Char(string='Description')
+    manual_ids = fields.One2many('ecdf.manual.input', 'report_id',
+        string="Manual Input(s)",
+        default=lambda self: self._default_manual())
+    language = fields.Selection(string='Language',
+        selection=[('FR', 'French'),
+                   ('DE', 'German'),
+                   ('EN', 'English')],
+        required=True,
+        default='FR')
+    type = fields.Selection(string='Declaration Type',
+        selection=[('month', 'Monthly'),
+                   ('quarter', 'Quarterly'),
+                   ('year', 'Annually')],
+        default='month',
+        required=True)
+    this_year = datetime.date.today().year
+    year = fields.Selection(string='Year',
+        selection=[(n, str(n)) for n in range(2019, this_year+1)],
+        required=True,
+        default=this_year)
+    period = fields.Selection(string="Period",
+        selection=[(1, 'January'),
+                   (2, 'February'),
+                   (3, 'March'),
+                   (4, 'April'),
+                   (5, 'May'),
+                   (6, 'June'),
+                   (7, 'July'),
+                   (8, 'August'),
+                   (9, 'September'),
+                   (10, 'October'),
+                   (11, 'November'),
+                   (12, 'December')],
+        required=True,
+        default=1)
+    agent_id = fields.Many2one(string='Agent', comodel_name='vat.agent', ondelete='restrict')
+    regime = fields.Selection(string='VAT accounting scheme',
+        selection=[('sales', 'On sales [204]'),
+                   ('revenues', 'On payments received [205]')],
+        required=True,
+        default='sales')
+
+    # File name (computed)
+    file_name = fields.Char(sring='File name', size=24, compute='_compute_file_name')
+    mis_instance_id = fields.Many2one('mis.report.instance', string="Mis report instance")
```

Comment:
There is a mechanism in mis builder to create temporary instances, perhaps we should use it here.

https://github.com/OCA/l10n-belgium/blob/2986d13c2ac0330c148c3155fdfb4dde5168b80e/l10n_be_mis_reports/data/mis_report_vat.xml#L6-L15 is a mechanism to open the mis builder instance creation form as a wizard. It can't be used as is here. But the gist of it is a [temporary](https://github.com/OCA/mis-builder/blob/c3d0a2910bdcc3eb00e214745aa80c3797d9c8f1/mis_builder/models/mis_report_instance.py#L463) flag on mis report instance, which "hides" them from the main list and have them garbage collected periodically.

Reply:
I'll take a look to that. With the "trick" I made to update the locals_dict, the report can work from the standard MIS report menu.

Reply:
I add the "temporary" flag on the instance so that it will be hidden from the list views. It will therefore be removed after 24h (which is not an issue knowing that the "preview" will create it again if needed).